store/storetest/channel_store.go

```diff
@@ -6617,13 +6618,287 @@ func testSidebarChannelsMigration(t *testing.T, ss store.Store) {
+	})
+
+	t.Run("should return any orphaned DM channels with the Direct Messages category", func(t *testing.T) {
+		user := &model.User{Id: model.NewId()}
+		teamId := model.NewId()
+
+		// Create the initial categories and find the DMs category
+		appErr := ss.Channel().CreateInitialSidebarCategories(user, teamId)
+		require.Nil(t, appErr)
+
+		categories, appErr := ss.Channel().GetSidebarCategories(user.Id, teamId)
+		require.Nil(t, appErr)
+		require.Equal(t, model.SidebarCategoryDirectMessages, categories.Categories[2].Type)
+
+		dmsCategory := categories.Categories[2]
+
+		// Create a DM
+		otherUserId := model.NewId()
+		dmChannel, err := ss.Channel().SaveDirectChannel(
+			&model.Channel{
+				Name: model.GetDMNameFromIds(user.Id, otherUserId),
+				Type: model.CHANNEL_DIRECT,
+			},
+			&model.ChannelMember{
+				UserId:      user.Id,
+				NotifyProps: model.GetDefaultChannelNotifyProps(),
+			},
+			&model.ChannelMember{
+				UserId:      otherUserId,
+				NotifyProps: model.GetDefaultChannelNotifyProps(),
+			},
+		)
+		require.Nil(t, err)
+
+		// Ensure that the DM is returned
+		res, appErr := ss.Channel().GetSidebarCategory(dmsCategory.Id)
+		assert.Nil(t, appErr)
+		assert.Equal(t, dmsCategory.Id, res.Id)
+		assert.Equal(t, model.SidebarCategoryDirectMessages, res.Type)
+		assert.Equal(t, []string{dmChannel.Id}, res.Channels)
+	})
+
+	t.Run("should return any orphaned GM channels with the Direct Messages category", func(t *testing.T) {
+		user := &model.User{Id: model.NewId()}
+		teamId := model.NewId()
+
+		// Create the initial categories and find the DMs category
+		appErr := ss.Channel().CreateInitialSidebarCategories(user, teamId)
+		require.Nil(t, appErr)
+
+		categories, appErr := ss.Channel().GetSidebarCategories(user.Id, teamId)
+		require.Nil(t, appErr)
+		require.Equal(t, model.SidebarCategoryDirectMessages, categories.Categories[2].Type)
+
+		dmsCategory := categories.Categories[2]
+
+		// Create a GM
+		gmChannel, err := ss.Channel().Save(&model.Channel{
+			Name:   "abc",
+			TeamId: "",
+			Type:   model.CHANNEL_GROUP,
+		}, 10)
+		require.Nil(t, err)
+		_, appErr = ss.Channel().SaveMember(&model.ChannelMember{
+			UserId:      user.Id,
+			ChannelId:   gmChannel.Id,
+			NotifyProps: model.GetDefaultChannelNotifyProps(),
+		})
+		require.Nil(t, appErr)
+
+		// Ensure that the DM is returned
+		res, appErr := ss.Channel().GetSidebarCategory(dmsCategory.Id)
+		assert.Nil(t, appErr)
+		assert.Equal(t, dmsCategory.Id, res.Id)
+		assert.Equal(t, model.SidebarCategoryDirectMessages, res.Type)
+		assert.Equal(t, []string{gmChannel.Id}, res.Channels)
+	})
+}
+
 func testCreateInitialSidebarCategories(t *testing.T, ss store.Store) {
 	t.Run("should create initial favorites/channels/DMs categories", func(t *testing.T) {
 		user := &model.User{Id: model.NewId()}
 		teamId := model.NewId()
 
 		err := ss.Channel().CreateInitialSidebarCategories(user, teamId)
-		require.Nil(t, err)
+		assert.Nil(t, err)
```

Comment:
any reason for the change?

Reply:
It's just something I missed writing the tests. I tend to use `require` for test set up and `assert` for checking things that I actually care about in the test